multimodal/tests/unittests/others/test_object_detection.py

```diff
@@ -290,3 +332,71 @@ def test_mmdet_object_detection_fit_then_inference_coco(checkpoint_name):
     )
 
     pred = predictor.predict(test_path)  # test batch inference
+
+
+@pytest.mark.parametrize(
+    "checkpoint_name",
+    [
+        "yolov3_mobilenetv2_320_300e_coco",
+    ],
+)
+def test_mmdet_object_detection_fit_df(checkpoint_name):
+    data_dir = download_sample_dataset()
+
+    train_path = os.path.join(data_dir, "Annotations", "trainval_cocoformat.json")
+    test_path = os.path.join(data_dir, "Annotations", "test_cocoformat.json")
+    # Init predictor
+    predictor = MultiModalPredictor(
+        hyperparameters={
+            "model.mmdet_image.checkpoint_name": checkpoint_name,
+            "env.num_gpus": 1,
+        },
+        problem_type="object_detection",
+        sample_data_path=train_path,
+    )
+
+    train_df = from_coco_or_voc(train_path)
+    predictor.fit(
+        train_df,
+        hyperparameters={
+            "optimization.learning_rate": 2e-4,
+            "env.per_gpu_batch_size": 2,
+        },
+        time_limit=30,
+    )
+
+
+@pytest.mark.parametrize(
+    "checkpoint_name",
+    [
+        "yolov3_mobilenetv2_320_300e_coco",
+    ],
+)
+def test_mmdet_object_detection_evaluate_df(checkpoint_name):
+    data_dir = download_sample_dataset()
+
+    train_path = os.path.join(data_dir, "Annotations", "trainval_cocoformat.json")
+    test_path = os.path.join(data_dir, "Annotations", "test_cocoformat.json")
+    # Init predictor
+    predictor = MultiModalPredictor(
+        hyperparameters={
+            "model.mmdet_image.checkpoint_name": checkpoint_name,
+            "env.num_gpus": 1,
+        },
+        problem_type="object_detection",
+        sample_data_path=train_path,
+    )
+
+    train_df = from_coco_or_voc(train_path)
+    test_df = from_coco_or_voc(test_path)
+    predictor.fit(
+        train_df,
+        hyperparameters={
+            "optimization.learning_rate": 2e-4,
+            "env.per_gpu_batch_size": 2,
+        },
+        time_limit=30,
+    )
+
+    preds = predictor.predict(data=test_df)
+    results = predictor.evaluate(data=test_df)
```

Comment:
Ditto. Consider combining the two tests.

Reply:
yea that makes sense